packager/media/formats/webm/multi_segment_segmenter.cc

```diff
@@ -28,14 +28,25 @@ Status MultiSegmentSegmenter::FinalizeSegment(uint64_t start_timestamp,
   CHECK(cluster());
   RETURN_IF_ERROR(Segmenter::FinalizeSegment(
       start_timestamp, duration_timestamp, is_subsegment));
+  if (!is_subsegment) {
+    std::string segment_name =
+        GetSegmentName(options().segment_template, start_timestamp,
+                       num_segment_++, options().bandwidth);
+
+    if (!writer_->WriteToFile(segment_name)) {
+      LOG(ERROR) << "Failed creating file for webm segment.";
+    }
+  }
+
```

Comment:
I think this should be moved after `cluster()->Finalize()`.

Reply:
@kqyang  Thanks for the review. I kept the file operations and did a writer_->WriteToFile(segment_name) before Cluster->Finalize() because Cluster->Finalize() does the following:
 ```
if (writer_->Seekable()) {

    const int64_t pos = writer_->Position();
 
    if (writer_->Position(size_position_))
      return false;
 
    if (WriteUIntSize(writer_, payload_size(), 8))
      return false;
 
    if (writer_->Position(pos))
      return false;
  }
```
Do you think it makes sense to keep the file operations or modify Write in BufferMkvWriter to insert at a given position?

Reply:
No. BufferMkvWriter is not seekable (`bool BufferMkvWriter::Seekable()` should always return false), so the quoted code is not executed.

Reply:
Ok. I will update the tests so that the cluster size value which is initialized to "unknown" in Cluster::WriteClusterHeader() is not updated with the above code block when we send false from Seekable().